examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -88,27 +92,52 @@ namespace {
 // This file is being used by platforms other than Linux, so we need this check to disable related features since we only
 // implemented them on linux.
 #if CHIP_DEVICE_LAYER_TARGET_LINUX
+constexpr EndpointId kNetworkCommissioningEndpointForThread = 0;
+constexpr EndpointId kNetworkCommissioningEndpointForWiFi   = 1;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
 NetworkCommissioning::LinuxThreadDriver sLinuxThreadDriver;
-Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(0 /* Endpoint Id */, &sLinuxThreadDriver);
+Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(kNetworkCommissioningEndpointForThread,
+                                                                             &sLinuxThreadDriver);
 #endif
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
 NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
-Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(1 /* Endpoint Id */, &sLinuxWiFiDriver);
-#endif
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(kNetworkCommissioningEndpointForWiFi, &sLinuxWiFiDriver);
 #endif
+#endif // CHIP_DEVICE_LAYER_TARGET_LINUX
 } // namespace
 
 void ApplicationInit()
 {
 #if CHIP_DEVICE_LAYER_TARGET_LINUX && defined(ZCL_USING_LEVEL_CONTROL_CLUSTER_SERVER)
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    sThreadNetworkCommissioningInstance.Init();
+    if (LinuxDeviceOptions::GetInstance().mThread)
+    {
+        sThreadNetworkCommissioningInstance.Init();
+    }
+    else
 #endif
+    {
+        uint32_t nullFeatureMapValue = 0;
```

Comment:
This is not a valid feature map for the network commissioning cluster...  It sort of makes things work for our tests, but this is not spec-compliant, so ideally we would not do this in example apps that people will then copy.

What I would suggest we do instead is:
1. When both thread and wifi are supported, put thread on endpoint 0 and wifi on endpoint 3, where network commissioning is the only thing on that endpoint.
2. When only thread is supported, disable endpoint 3 (via `emberAfEndpointEnableDisable`), so it just doesn't show up at all.
3. When only wifi is supported, disable endpoint 3 and put wifi on endpoint 0.  I suspect we can do this by calling the destructor of `sWiFiNetworkCommissioningInstance` and then doing placement-new into `&sWiFiNetworkCommissioningInstance` with the new endpoint id.

Reply:
Moved the instance on endpoint 1 to endpoint 65534, and added a note about setting the value of endpoint 0

Reply:
> Maybe we can set it to Ethernet interface type? However in real products we should hide the cluster instance.

Ah, did not see this before my other comments.  A spec-compliant impl MUST have this cluster on EP 0.  See device library spec.

So yes, if we don't do wifi or thread we must be doing ethernet and should set to that type.

Reply:
After checking tests, I added a NullNetworkDriver which will disable the cluster at cluster level (always return false for "Enabled" attribute)